### PR TITLE
Move import cauchy into the function to avoid pickling error 

### PR DESCRIPTION
**Pull request by hopedisastro**
Move the import for the cauchy distribution into the function definition to avoid pickling error. 

https://centrepopgen.slack.com/archives/C030X7WGFCL/p1713737228278469?thread_ts=1713324735.266539&cid=C030X7WGFCL

### CODE REVIEW

**Comment by MattWellie** [comment]
can you run `ruff check --fix .` locally to pass the linting checks

File: str/associatr/multiple_testing_correction/run_gene_level_pval.py [hunk 1]
```diff
@@ -65,6 +64,7 @@ def cct(
     #' @export
     R code is implemented in python
     """
+    from scipy.stats import cauchy
```
Comment (jmarshall) [inline str/associatr/multiple_testing_correction/run_gene_level_pval.py:67]: It'd be worth putting a comment on this so it doesn't get accidentally “tidied up” in future… something like:

```suggestion
    # Import here as a PythonJob's function must stand alone
    from scipy.stats import cauchy  # noqa: PLC0415
```

(The `noqa` will stop ruff from complaining about this if the relevant check gets applied by default in future.)

**Comment by jmarshall** [comment]
If this is the problem, I wonder if you're going to have the same problem with `to_path` and `output_path`…

**Comment by hopedisastro** [comment]
Matt proposed an optimised runner (#180) because the current implementation was very inefficient in getting the driver job to complete when running using the full dataset using all cell types

**Comment by MattWellie** [comment]
Apologies, the diff now looks awful 😓 

- can you reduce the concurrency permitted to remove those errors you saw in test
- can you add an escape for 'this file already exists' to get rid of those preemption problems (concurrency cap might solve that anyway)

Happy to have a crack at those, or if you're happy to just forge ahead with the concurrency change

**Review by MattWellie — APPROVED** [review]
approved on the basis that the concurrency CLI argument will be reduced when used

File: str/associatr/multiple_testing_correction/run_gene_level_pval.py [hunk 2]
```diff
@@ -163,16 +215,8 @@ def bonferroni_compute(
     default=500,
```
Comment (hopedisastro) [inline str/associatr/multiple_testing_correction/run_gene_level_pval.py:215]:
```suggestion
    default=18,
```

**Pull request merged**